File: inserccion.py
from time import clock

lista = [3,1,8,2,5]

tiempo_inicial = clock()
for i in range(1,len(lista)):
	temp = lista[i]
	j = i -1
	while j >= 0:
		if temp < lista[j]:
			lista[j+1] = lista[j]
			lista[j] = temp
			j = j-1
		else:
			break
tiempo_final = clock()
			
print (lista)
print (tiempo_final - tiempo_inicial)

File: inserccion.cpp
#include <iostream>
#include <time.h>

using namespace std;

int main()
{
		int tam = 5;
		int lista[tam] = {3,1,8,2,5};
		int temp,i,j;

		for (i=1; i<tam; i++){
			 temp = lista[i];
			 for (j=i-1; j>=0 && lista[j] > temp; j--){
			 	lista[j+1]=lista[j];
			 }
			 lista[j+1]=temp;
			 }

		for (i=0; i<tam; i++){
			cout <<lista[i] << " ";
		}

		system("pause");
		return 0;
	}
